src/modeci_mdf/mdf.py

```diff
@@ -96,14 +96,19 @@ def _structure(cls, o: Dict[str, Any], t: Any) -> "Function":
         except KeyError:
             args = {}
 
+        try:
+            metadata = o["metadata"]
+        except KeyError:
+            metadata = {}
```

Comment:
This looks ok to me. But maybe to avoid this problem creeping up going forward (new attributes added to base class) we can do something like this:

```
base_class_args = {f.name: o[f.name] for f in attr.fields(cls.__bases__[0]) if f.name in o}
```
and then pass ```**base_clas_kwargs``` to the constructors. I haven't tested this though so if it doesn't work just merge this for now I guess. This whole override to handle these translated versions of the function specification thing I feel is an unneeded hack. We are essentially rewriting cattrs functionality here just because we want to handle these alternative function specifications.

Reply:
Hm, ok, so do you think this problem is probably happening in other classes or attribute too? And this change would be something for MdfBase or Base?

Reply:
I don't think so, most of the other classes should use the cattr generated version of these hooks. Maybe Parameter doesn't cause it is another weird one if I remember correctly. I needed to override cattr hook for Functions because of the multiple ways in which we allow them to be specified in JSON. This mostly comes from the MDF "translation" examples. All those ways need to get consolidated to the single Python object representation when structuring.

Reply:
I see. You're right about the translation examples - they're also the only places I see where no `id` is present when constructing a Function (id is listed as a required attr). It's odd though, because a parameter-as-function specified similarly doesn't need to explicitly set id, it seems auto detected from the dictionary key. Like this example assigns level as the id for this Parameter.
https://github.com/ModECI/MDF/blob/b718d7484bf72f45ee0796e1ec4d9ac050f192cd/examples/MDF/RNN/RNNs.json#L24-L30

Is this also related to overriding the cattr hook? If so, would it be possible to call the original hook in our overriding version? The intent being to just modify the function-args dictionary format and then keep the rest of the default hook behavior.

Reply:
I agree it's fine to add this as a quick fix for now, but it should be resolved to make sure the function specification is completely fixed by MDF definitions, and automatically handled by attrs/cattrs. 
Some of the changes made in the translated/MDF0 examples were still a bit up in the air, and may not be specified in the best way. 

For this reason it would be even more important to have one representative example looking at all the forms allowable for parameters/functions, see if there is too much overlap, and see if we can manage with a minimal set.